src/Tree/Strategy/ORM/Closure.php

```diff
@@ -271,8 +271,14 @@ public function processPostPersist($em, $entity, AdapterInterface $ea)
     {
         $uow = $em->getUnitOfWork();
         $emHash = spl_object_id($em);
+        $entityClass = get_class($entity);
 
-        while ($node = array_shift($this->pendingChildNodeInserts[$emHash])) {
+        foreach ($this->pendingChildNodeInserts[$emHash] as $node) {
+            $nodeClass = get_class($node);
+            if ($entityClass !== $nodeClass) {
+                continue;
+            }
+            unset($this->pendingChildNodeInserts[$emHash][spl_object_id($node)]);
```

Comment:
IMO, a comment explaining this "reset" could help to achieve a better understanding of the flow.

Reply:
I've added a comment. Let me know if you think it provides enough explanation of why some nodes are being skipped

Reply:
IMO, that's fine for the skipped iterations, but I'd still like to have a comment about this call to `reset()`.

Reply:
Thanks for clarifying. I've added a comment for the call to unset(). Let me know if that gives enough clarity.

Reply:
I think this is enough. Thank you!